src/components/common/store/bookmarkListReducer.tsx

```diff
@@ -141,12 +140,41 @@ export const {
 
 export { on, off };
 
+export const checkExtentAndAdd = createAsyncThunk<
+  void,
+  OGCCollection,
+  { rejectValue: ErrorResponse; state: RootState; dispatch: AppDispatch }
+>("bookmarkList/checkExtentAndAdd", async (item, thunkAPI) => {
+  const { dispatch } = thunkAPI;
+
+  // Check if the item has a valid extent
+  const hasValidExtent = checkExtent(item);
+
+  if (hasValidExtent) {
+    // If the item has a valid extent, add it directly
+    dispatch(addItem(item));
+  } else {
+    // If the item does not have a valid extent, fetch it first
+    await dispatch(fetchResultByUuidNoStore(item.id))
+      .unwrap()
+      .then((res: OGCCollection) => {
+        dispatch(addItem(res));
+      })
+      .catch((err: Error) => {
+        errorHandling(thunkAPI);
+      });
+  }
+});
+
 // Thunk actions for async operations, avoid getState() where someone is calling the reducer
-export const fetchAndInsertTemporary = createAsyncThunk<
+export const checkAndInsertTemporary = createAsyncThunk<
```

Comment:
Do not like to do it this way, it is better to have the caller component to sort that out for you instead of backfill something here, same for checkAndInsertTemporary, can you redo without adding two functions?